sdk-dotnet/LittleHorse.Sdk/Internal/LHLoggerFactoryProvider.cs

```diff
@@ -0,0 +1,22 @@
+using LittleHorse.Sdk.Exceptions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace LittleHorse.Sdk.Internal
+{
+    public static class LHLoggerFactoryProvider
+    {
+        private static ILoggerFactory? _loggerFactory;
+
+        public static void Initialize(ILoggerFactory? loggerFactory = null)
+        {
+            _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
+        }
+
+        public static ILogger<T> GetLogger<T>()
+        {
+            if (_loggerFactory != null) return _loggerFactory.CreateLogger<T>();
+            throw new LHLoggerFactoryNullException("_loggerFactory does not have a valid value and it is trying to create a logger instance.");
```

Comment:
I think you can use [InvalidOperationException](https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-8.0) instead of `LHLoggerFactoryNullException`

Reply:
[Suggestion]
Use if/else structure. IMO, it is cleaner.

Reply:
sure, let me see